src/test/java/test/configuration/BeforeMethodWithGroupFiltersTest.java

```diff
@@ -0,0 +1,21 @@
+package test.configuration;
+
+import org.testng.TestNG;
+import org.testng.annotations.Test;
+import test.SimpleBaseTest;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class BeforeMethodWithGroupFiltersTest extends SimpleBaseTest {
+
+    @Test
+    public void beforeMethodWithBeforeGroupsShouldOnlyRunBeforeGroupMethods() {
+        TestNG suite = create(new Class[] { BeforeMethodWithGroupFiltersSampleTest.class } );
+        suite.run();
+        for (String[] expectedSequence : BeforeMethodWithGroupFiltersSampleTest.EXPECTED_INVOCATION_SEQUENCES)
+            assertThat(BeforeMethodWithGroupFiltersSampleTest.invocations)
```

Comment:
the assertion is not strong enough: the order should not change so you can provide the exact list of invoked methods.